apiserver/facades/client/cloud/cloud_test.go

```diff
@@ -898,19 +1163,50 @@ func (s *cloudSuite) TestUpdateCredentialsSomeModelsFailedValidation(c *gc.C) {
 }
 
 func (s *cloudSuite) TestUpdateCredentialsSomeModelsFailedValidationForce(c *gc.C) {
-	s.backend.credentialModelsF = func(tag names.CloudCredentialTag) (map[string]string, error) {
-		return map[string]string{
-			coretesting.ModelTag.Id():              "testModel1",
-			"deadbeef-2f18-4fd2-967d-db9663db7bea": "testModel2",
-		}, nil
+	adminTag := names.NewUserTag("admin")
+	defer s.setup(c, adminTag).Finish()
+
+	model := map[string]string{
+		coretesting.ModelTag.Id():              "testModel1",
+		"deadbeef-2f18-4fd2-967d-db9663db7bea": "testModel2",
 	}
 
-	s.PatchValue(cloudfacade.ValidateNewCredentialForModelFunc, func(backend credentialcommon.PersistentBackend, callCtx context.ProviderCallContext, credentialTag names.CloudCredentialTag, credential *cloud.Credential, migrating bool) (params.ErrorResults, error) {
-		if backend.(*mockModelBackend).uuid == "deadbeef-0bad-400d-8000-4b1d0d06f00d" {
-			return params.ErrorResults{[]params.ErrorResult{{&params.Error{Message: "not valid for model"}}}}, nil
+	aCloud := jujucloud.Cloud{
+		Name:      "dummy",
+		Type:      "dummy",
+		AuthTypes: []jujucloud.AuthType{jujucloud.EmptyAuthType, jujucloud.UserPassAuthType},
+		Regions:   []jujucloud.Region{{Name: "nether", Endpoint: "endpoint"}},
+	}
+
+	getF := func(modelUUID string) (credentialcommon.PersistentBackend, context.ProviderCallContext, error) {
+		if modelUUID == "deadbeef-0bad-400d-8000-4b1d0d06f00d" {
+			return newModelBackend(c, aCloud,
+				modelUUID), context.NewEmptyCloudCallContext(), errors.New("not valid for model")
 		}
-		return params.ErrorResults{[]params.ErrorResult{}}, nil
-	})
+		return newModelBackend(c, aCloud, modelUUID), context.NewEmptyCloudCallContext(), nil
+	}
+
+	tag, err := cloudCredentialTag("three", "julia", "meep", map[string]string{}).CloudCredentialTag()
+	c.Assert(err, jc.ErrorIsNil)
+
+	s.PatchValue(cloudfacade.ValidateNewCredentialForModelFunc,
+		func(
+			backend credentialcommon.PersistentBackend, callCtx context.ProviderCallContext,
+			credentialTag names.CloudCredentialTag, credential *jujucloud.Credential, migrating bool,
+		) (params.ErrorResults, error) {
+			if backend.(*mockModelBackend).uuid == "deadbeef-0bad-400d-8000-4b1d0d06f00d" {
+				return params.ErrorResults{[]params.ErrorResult{{&params.Error{Message: "not valid for model"}}}}, nil
+			}
+			return params.ErrorResults{[]params.ErrorResult{}}, nil
+		})
```

Comment:
This patch is done multiple times, in very similar ways (this being the biggest exception). Would it be possible to extract this out in some form into `SetUpTest` or `setup`?